apps/gallery/test/marionette/activity_test.js

```diff
@@ -33,128 +33,165 @@ marionette('Gallery Activity Tests', function() {
+      activityTesterApp.launch();
+      // Initiate open activity by clicking open image button.
+      // This will generate memory backed blob using canvas
+      // and trigger open activity to view the blob
+      activityTesterApp.tapOpenImageButton();
 
+      system.switchToApp(Gallery.ORIGIN);
 
-  test('open memory backed blob using gallery open activity', function() {
-    activityTesterApp.launch();
-    // Initiate open activity by clicking open image button.
-    // This will generate memory backed blob using canvas
-    // and trigger open activity to view the blob
-    activityTesterApp.tapOpenImageButton();
+      // Check there is no filename in title bar
+      var title = fullscreenView.openActivityImageTitle.text();
+      assert.strictEqual(title, '');
 
-    system.switchToApp(Gallery.ORIGIN);
+      // Check save button is not displayed
+      assert.ok(!fullscreenView.openActivitySaveButton.displayed(),
+        'Save button is not displayed');
 
-    // Check there is no filename in title bar
-    var title = fullscreenView.openActivityImageTitle.text();
-    assert.strictEqual(title, '');
+      client.waitFor(function(){
+        return fullscreenView.openActivityImage.displayed();
+      });
 
-    // Check save button is not displayed
-    assert.ok(!fullscreenView.openActivitySaveButton.displayed(),
-      'Save button is not displayed');
+      // Check if displayed image src is set with blob url
+      assert.ok(fullscreenView.hasSrcImageBlobURL(Gallery.ORIGIN,
+        fullscreenView.openActivityImage));
+     });
+  });
 
-    client.waitFor(function(){
-      return fullscreenView.openActivityImage.displayed();
+  suite('Gallery app multiple images', function() {
+    setup(function() {
+      // Remove all files in temp device storage.
+      client.fileManager.removeAllFiles();
+      // Add file into the pictures directory
+      client.fileManager.add([
+        {type: 'pictures',filePath: 'test_media/Pictures/firefoxOS.png'},
+        {type: 'pictures',
+         filePath: 'test_media/Pictures/firefoxOS.png',
+         filename: 'firefoxOS_2.png'}
+      ]);
     });
 
-    // Check if displayed image src is set with blob url
-    assert.ok(fullscreenView.hasSrcImageBlobURL(Gallery.ORIGIN,
-      fullscreenView.openActivityImage));
-   });
+    test('SelectView > share image using share activity ', function() {
+      galleryApp.launch();
+      galleryApp.switchToSelectView();
+
+      // Check number of files that exist
+      assert.ok(galleryApp.thumbnails.length == '2');
+      galleryApp.tapThumbnail(0);
+
+      galleryApp.thumbnailsShareButton.tap();
+      system.menuOptionButton('Gallery Activity Tester').tap();
+      system.switchToApp(GalleryActivityTester.ORIGIN);
+
+      // Check that image is shared and displayed in
+      // Gallery Activity Tester App
+      client.waitFor(function(){
+        return activityTesterApp.sharedImage.displayed();
+      });
+    });
```

Comment:
In the other share test, you check that the image name and type and size are as expected. Can you do that here, too?  Maybe putting the code shared by the two tests into a utility function?